### DISCUSSION TITLE
data property is undefined although my source is not

### DISCUSSION THREAD

**Discussion by rpimentaf** [komarovalexander/ka-table#206] [Q&A]
Hi!
My code is:
```
const dataArray = data?.result.nodes.map((x) => ({
       id: x.id,
       iataOrigem: x.siglaIataAerodromoOrigem,
       iataDestino: x.siglaIataAerodromoDestino,
       iataOperador: x.siglaIataEmpresaAerea,
       nrVoo: x.nrVoo,
       partida: x.dataHoraPartidaLocal,
       chegada: x.dataHoraChegadaLocal
   }));
const tablePropsInit: ITableProps = {
      columns: [
         {
            dataType: DataType.Number,
            key: 'id',
            style: {textAlign: 'center'},
            title: 'Id',
         },
         {
            dataType: DataType.String,
            key: 'iataOrigem',
            style: {textAlign: 'center'},
            title: 'Iata Origem',
         },
         {
            dataType: DataType.String,
            key: 'iataDestino',
            style: {textAlign: 'center'},
            title: 'Iata Destino',
         },
         {
            dataType: DataType.String,
            key: 'iataOperador',
            style: { textAlign: 'center' },
            title: 'Iata Operador',
         },
         {
            dataType: DataType.String,
            key: 'nrVoo',
            style: { textAlign: 'center' },
            title: 'Nr. Voo',
         },
         {
            dataType: DataType.Date,
            key: 'partida',
            style: { textAlign: 'center' },
            title: 'Partida',
         },
         {
            dataType: DataType.Date,
            key: 'chegada',
            style: { textAlign: 'center' },
            title: 'Chegada',
         }
      ],
      format: ({ column, value }) => {
         if (column.dataType === DataType.Date) {
            return (
               value &&
               value.toLocaleDateString('pt-br', {
                  month: '2-digit',
                  day: '2-digit',
                  year: 'numeric',
                  hour: '2-digit',
                  minute: '2-digit'
              })
            );
         } 
         return value;         
      },
      data: dataArray,
      rowKeyField: 'id'
   };
```
The result of dataArray in console is an array of length 10:
`(10) [{…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}]`

So, why the tableProps has data property undefined, as shown in console:
`columns: (7) [{…}, {…}, {…}, {…}, {…}, {…}, {…}]
data: undefined
rowKeyField: "id"`

Thanks for your help!!!

**Comment by komarovalexander** [comment]
Hi @rpimentaf can you reproduce it on https://stackblitz.com/edit/table-overview-ts?file=Demo.tsx ?

**Reply by rpimentaf** [reply]
@komarovalexander, tks for your answer.
I've authorized you to see my profile in Table Overview. Besides this, look at this pic...
![console](https://user-images.githubusercontent.com/69866458/175792843-33a956b4-31f4-4e0f-a286-92be1907bf62.png)

Although the Demo environment works, my code does not.

My entire code is:
[index.txt](https://github.com/komarovalexander/ka-table/files/8985704/index.txt)

Any suggestions???

Thanks very much!!!!

**Comment by komarovalexander** [comment]
TablePropsInit should contain only static values. I would suggest to define it outside of component. 
use updateData to update the array after the request https://komarovalexander.github.io/ka-table/#/remote-data

**Comment by rpimentaf** [comment]
@komarovalexander , thank you for your answer. Since then i'm struggling to have a solution, but untill now i don't have it. But i know that you gave me the path to have it.
So, in other words, thank you for your help and congratulations for your well done job.
